### PR TITLE
update pipeline-service ci setup

### PR DESCRIPTION
**Pull request by xinredhat**
For now CI job switched to use Hyperhift Cli for creating and deleting Hypershift hosted cluster. This PR updated the scripts for the process of Hypershift Setup.

 Signed-off-by: Xin Jiang @xinredhat

### CODE REVIEW

File: ci/docs/continuous_integation.md [hunk 1]
```diff
@@ -10,13 +10,15 @@ HyperShift official [Documentation Guide](https://hypershift-docs.netlify.app/).
...
-- Install [aws](https://docs.aws.amazon.com/cli/latest/userguide/getting-started-install.html) cli
+- Install [AWS](https://docs.aws.amazon.com/cli/latest/userguide/getting-started-install.html) cli
+- Install [HyperShift](https://hypershift-docs.netlify.app/getting-started/#prerequisites) cli
 - Provision a ROSA cluster by following [documentation](rosa_cluster_provision.md)
 
 ### HyperShift Setup
-
 You can configure HyperShift on ROSA by running the [hypershift_setup.sh](../hack/hypershift_setup.sh) script.
 
+After that, you need to add a Configmap for storing the kubeconfig of ROSA cluster and a Secret for sotring the Bitwarded crednetials(BW_CLIENTID,BW_CLIENTSECRET and BW_PASSWORD).
```
Comment (Roming22) [inline ci/docs/continuous_integation.md:20]: `storing`, `Bitwarden`, `credentials`

File: ci/docs/continuous_integation.md [hunk 2]
```diff
@@ -10,13 +10,15 @@ HyperShift official [Documentation Guide](https://hypershift-docs.netlify.app/).
 
 ### Pre-requisites:
 
-- Install [aws](https://docs.aws.amazon.com/cli/latest/userguide/getting-started-install.html) cli
+- Install [AWS](https://docs.aws.amazon.com/cli/latest/userguide/getting-started-install.html) cli
```
Comment (Roming22) [inline ci/docs/continuous_integation.md:13]: `aws`, as this is the command line binary, not the service.

File: ci/docs/continuous_integation.md [hunk 3]
```diff
@@ -10,13 +10,15 @@ HyperShift official [Documentation Guide](https://hypershift-docs.netlify.app/).
...
+- Install [HyperShift](https://hypershift-docs.netlify.app/getting-started/#prerequisites) cli
```
Comment (Roming22) [inline ci/docs/continuous_integation.md:14]: `hypershift`, as this is the cli binary, not the product.

File: ci/hack/hypershift_setup.sh [hunk 4]
```diff
@@ -125,55 +95,46 @@ prechecks() {
     fi
 }
 
-init() {
-  SCRIPT_DIR=$(
-    cd "$(dirname "$0")" >/dev/null
-    pwd
-  )
+create_s3_bucket() {
+  #check if the s3 bucket is there
```
Comment (Roming22) [inline ci/hack/hypershift_setup.sh:99]: `# Check`

File: ci/hack/hypershift_setup.sh [hunk 5]
```diff
@@ -125,55 +95,46 @@ prechecks() {
...
-  kubectl create ns ci-clusters
-
-  oc create secret generic my-aws-cred -n ci-clusters \
-    --from-literal=baseDomain="$HYPERSHIFT_BASE_DOMAIN" \
-    --from-literal=aws_access_key_id="$AWS_ACCESS_KEY_ID" \
-    --from-literal=aws_secret_access_key="$AWS_SECRET_ACCESS_KEY" \
-    --from-literal=pullSecret="$HYPERSHIFT_PULL_SECRET" \
-    --from-file=ssh-publickey="$HOME/.ssh/id_rsa.pub" \
-    --from-file=ssh-privatekey="$HOME/.ssh/id_rsa"
+  # Retrive AWS Credential file from Bitwarden
```
Comment (Roming22) [inline ci/hack/hypershift_setup.sh:112]: `# Retrieve`

File: ci/hack/hypershift_setup.sh [hunk 6]
```diff
@@ -125,55 +95,46 @@ prechecks() {
...
+  open_bitwarden_session
+  get_aws_credentials
+  
+  # Install Hypershift operator
```
Comment (Roming22) [inline ci/hack/hypershift_setup.sh:116]: `HyperShift`

File: ci/hack/hypershift_setup.sh [hunk 7]
```diff
@@ -125,55 +95,46 @@ prechecks() {
...
+  hypershift install --oidc-storage-provider-s3-credentials "$AWS_CREDENTIALS" \
+    --oidc-storage-provider-s3-bucket-name "$BUCKET_NAME" \
+    --oidc-storage-provider-s3-region="$BUCKET_REGION"
+
+  #a loop to check if the deployment is Available and Ready
```
Comment (Roming22) [inline ci/hack/hypershift_setup.sh:121]: `# Loop to`

File: ci/hack/hypershift_setup.sh (hunk 6 above)
Comment (xinredhat) [reply ci/hack/hypershift_setup.sh]: fixed

File: ci/hack/hypershift_setup.sh (hunk 7 above)
Comment (xinredhat) [reply ci/hack/hypershift_setup.sh]: fixed

File: ci/hack/hypershift_setup.sh (hunk 5 above)
Comment (xinredhat) [reply ci/hack/hypershift_setup.sh]: fixed

File: ci/hack/hypershift_setup.sh (hunk 4 above)
Comment (xinredhat) [reply ci/hack/hypershift_setup.sh]: fixed

File: ci/docs/continuous_integation.md (hunk 3 above)
Comment (xinredhat) [reply ci/docs/continuous_integation.md]: fixed

File: ci/docs/continuous_integation.md (hunk 2 above)
Comment (xinredhat) [reply ci/docs/continuous_integation.md]: fixed

File: ci/docs/continuous_integation.md (hunk 1 above)
Comment (xinredhat) [reply ci/docs/continuous_integation.md]: fixed

**Review by Roming22 — APPROVED** [review]
If the PR is ready to be merged, please remove the `WIP` from the PR title.

**Comment by xinredhat** [comment]
> If the PR is ready to be merged, please remove the `WIP` from the PR title.

not yet, I need to verify the scripts locally

**Pull request merged**